tracing-error/src/error.rs

```diff
@@ -60,6 +60,75 @@ pub struct TracedError<E> {
     inner: ErrorImpl<E>,
 }
 
+impl<E> TracedError<E>
+where
+    E: std::error::Error + Send + Sync + 'static,
+{
+    /// Convert the inner error type of a `TracedError` while preserving the
+    /// existing SpanTrace.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use tracing_error::TracedError;
+    /// # #[derive(Debug)]
+    /// # struct InnerError;
+    /// # #[derive(Debug)]
+    /// # struct OuterError(InnerError);
+    /// # impl std::fmt::Display for InnerError {
+    /// #     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    /// #         write!(f, "Inner Error")
+    /// #     }
+    /// # }
+    /// # impl std::error::Error for InnerError {
+    /// # }
+    /// # impl std::fmt::Display for OuterError {
+    /// #     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    /// #         write!(f, "Outer Error")
+    /// #     }
+    /// # }
+    /// # impl std::error::Error for OuterError {
+    /// #     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+    /// #         Some(&self.0)
+    /// #     }
+    /// # }
+    ///
+    /// let err: TracedError<InnerError> = InnerError.into();
+    /// let err: TracedError<OuterError> = err.map(|inner| OuterError(inner));
+    /// ```
+    pub fn map<F, O>(self, op: O) -> TracedError<F>
+    where
+        O: FnOnce(E) -> F,
```

Comment:
nit, take it or leave it: since the return type doesn't depend on the actual _type_ of the mapped function (it's applied immediately), we could just use `impl FnOnce` here, which might be a little more concise?
```suggestion
    pub fn map<F>(self, op: impl FnOnce(E) -> F) -> TracedError<F>
    where
```

Reply:
I'm a little worried about using `impl Trait` syntax here because it disables turbofish, which might be useful when dealing with type inference